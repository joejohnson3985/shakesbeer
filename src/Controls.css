.filter-area {
  display: flex;
  align-items: center;
}

#fav-filter {
  width: 40px;
  padding: 10px;
  text-align: center;
}