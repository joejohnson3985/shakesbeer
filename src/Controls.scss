.filter-area {
  display: flex;
  padding: 20px 0;
  justify-content: space-between;
  align-items: center;
  width: 1000px;
}

.fav-filter-area {
  text-align: right;
}

#fav-filter {
  width: 40px;
  padding: 10px;
  text-align: center;
}