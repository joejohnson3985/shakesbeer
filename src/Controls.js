import React, { Component } from 'react';
import Filter from './Filter.js';
import './Controls.css';

class Controls extends Component {
  constructor(props) {
    super(props);
    this.state = {
    }
  }

  updateCity = (e) => {
    this.props.updateFilterSelection('city', e.target.value); 
  }

  updateStyle = (e) => {
    this.props.updateFilterSelection('style', e.target.value); 
  }

  updateIbu = (e) => {
    this.props.updateFilterSelection('ibu', e.target.value); 
  }

  render() {
    return (
      <form className="filter-area">
        <Filter 
          className="city-filter"
          filterName="city-filter"
          label="City: "
          filterOptions={this.props.breweryCities}
          updateSelected={this.updateCity}
        />
        <Filter
          className="style-filter" 
          filterName="style-filter" 
          label="Style: " 
          filterOptions={this.props.beerStyles} 
          updateSelected={this.updateStyle}
        />
        <Filter 
          className="ibu-filter" 
          filterName="ibu-filter" 
          label="Ibu: " 
          filterOptions={this.props.beerIbus} 
          updateSelected={this.updateIbu}
        />
        {/* Add Filter for abv */}

        {/* <label htmlFor='city-filter' className='filter-label'>City: </label>
        <select className='filter' id='city-filter' onChange={this.updateCity}>
        <option>All</option>
          {
            this.props.breweryCities.map(loc => {
              return(
                <option className='state-dropdown-options' key={loc}>{loc}</option>
              )
            })
          }
        </select>
        <label htmlFor='style-filter' className='filter-label'>Style: </label>
        <select className='filter' id='style-filter' onChange={this.updateStyle}>
        <option>All</option>
          {
            this.props.beerStyles.map(loc => {
              return(
                <option className='state-dropdown-options' key={loc}>{loc}</option>
              )
            })
          }
        </select>
        <label htmlFor='ibu-filter' className='filter-label'>ibu: </label>
        <select className='filter' id='ibu-filter' onChange={this.updateIbu}>
        <option>All</option>
          {
            this.props.beerIbus.map(loc => {
              return(
                <option className='state-dropdown-options' key={loc}>{loc}</option>
              )
            })
          }
        </select> */}
      </form>
    )
  }
}

  export default Controls;