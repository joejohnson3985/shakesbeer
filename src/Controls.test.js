import React from 'react';
import Controls from './Controls.js';
import { shallow } from 'enzyme';

const mockData = [1,2,3,4]
const mockFunc = jest.fn()

describe('Controls', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow (
            <Controls breweryCities={mockData} 
                      beerStyles={mockData}
                      filterByCity={mockFunc}
                      filterSelections={mockData}
                      updateFilterSelection={mockFunc}
            />
        )
    });

    it('Should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    });

    // it('Should simulate an onChange event', () => {
    //     wrapper.find('#city-filter').simulate('change', {target:{value: 'e'}});
    //     expect(mockFunc).toBeCalled();
    // });

    // it('Should simulate an onChange event', ()=> {
    //     wrapper.find('#style-filter').simulate('change', {target:{value: 'e'}});
    //     expect(mockFunc).toBeCalled();
    // });
});