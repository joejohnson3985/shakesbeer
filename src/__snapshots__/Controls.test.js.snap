// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls Should match the snapshot 1`] = `
<form
  className="filter-area"
>
  <Filter
    filterName="city-filter"
    filterOptions={
      Array [
        1,
        2,
        3,
        4,
      ]
    }
    label="City: "
    updateSelected={[Function]}
  />
  <Filter
    filterName="style-filter"
    filterOptions={
      Array [
        1,
        2,
        3,
        4,
      ]
    }
    label="Style: "
    updateSelected={[Function]}
  />
  <Filter
    filterName="ibu-filter"
    filterOptions={[MockFunction]}
    label="Ibu: "
    updateSelected={[Function]}
  />
  <div
    className="fav-filter-area"
  >
    <label
      htmlFor="fav-filter"
    >
      Starred Only: 
    </label>
    <input
      id="fav-filter"
      type="checkbox"
    />
  </div>
</form>
`;
